packages/flutter/test/services/message_codecs_test.dart

```diff
@@ -18,26 +16,25 @@ import 'message_codecs_testing.dart';
 
 void main() {
   group('Binary codec', () {
-    const MessageCodec<ByteData> binary = BinaryCodec();
+    const MessageCodec<ByteData?> binary = BinaryCodec();
     test('should encode and decode simple messages', () {
-      checkEncodeDecode<ByteData>(binary, null);
-      checkEncodeDecode<ByteData>(binary, ByteData(0));
-      checkEncodeDecode<ByteData>(binary, ByteData(4)..setInt32(0, -7));
+      checkEncodeDecode<ByteData?>(binary, null);
+      checkEncodeDecode<ByteData?>(binary, ByteData(0));
+      checkEncodeDecode<ByteData?>(binary, ByteData(4)..setInt32(0, -7));
     });
   });
   group('String codec', () {
-    const MessageCodec<String> string = StringCodec();
+    const MessageCodec<String?> string = StringCodec();
```

Comment:
I think we should not use nullable types other than L21.
`MessageCodec.encodeMessage` takes `T` instead of `T?`. I think in most cases `T` is a non-nullable type, which means `encodeMessage` will seldom get a null. If we really want to make sure that `MessageCodec` does not crash with nullable types, L21 is sufficient.

Reply:
Same goes to platform_channel_test

Reply:
Well, but the problem is that the `StringCodec` is defined as `MessageCodec<String?>` and `BinaryCodec` is defined as a `MessageCodec<ByteData?>`, and the implementation does want to be able to return null for messages that are null. I tried making `MessageCodec<String>` be the base of `StringCodec`, but that had a cascade effect on anything using messages, and is a much bigger PR. I'm not even sure it's the right thing: there may be places where having a null message makes sense. Applies to `platform_channel_test` too.

Reply:
That makes sense. Thanks!